src/System.Private.CoreLib/src/Internal/Reflection/Core/NonPortable/RuntimeType.cs

```diff
@@ -302,12 +288,47 @@ public override RuntimeTypeHandle TypeHandle
         public abstract String InternalFullNameOfAssembly { get; }
 
         //
+        // NOTE: If you override this, you must also override InternalTryGetEETypePtr.
+        //
+        public virtual EETypePtr ToEETypePtr()
+        {
+            if (this.IsByRef)
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_NoTypeHandleForByRef);
+
+            // If a constructed type doesn't have an type handle, it's either because the reducer tossed it (in which case,
+            // we would thrown a MissingMetadataException when attempting to construct the type) or because one of
+            // component types contains open type parameters. Since we eliminated the first case, it must be the second.
+            // Throwing PlatformNotSupported since the desktop does, in fact, create type handles for open types.
+            if (this.HasElementType || this.IsConstructedGenericType || this.IsGenericParameter)
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_NoTypeHandleForOpenTypes);
+
+            // If got here, this is a "plain old type" that has metadata but no type handle. We can get here if the only
+            // representation of the type is in the native metadata and there's no EEType at the runtime side.
+            // If you squint hard, this is a missing metadata situation - the metadata is missing on the runtime side - and
+            // the action for the user to take is the same: go mess with RD.XML.
+            throw RuntimeAugments.Callbacks.CreateMissingMetadataException(this);
+        }
+
+        //
+        // NOTE: If you override this, you must also override ToEETypePtr.
         // (Note: I don't like this member being public but I can't use "internal" since this is shared across two implementation assemblies.)
-        // NOTE: If you override this, you must also override TypeHandle.
         //
-        public virtual bool InternalTryGetTypeHandle(out RuntimeTypeHandle typeHandle)
+        public virtual bool InternalTryGetEETypePtr(out EETypePtr eeType)
         {
-            typeHandle = default(RuntimeTypeHandle);
+            eeType = default(EETypePtr);
+            return false;
+        }
+
+        // NOTE: This will only succeed if the type has an underlying EEType
+        public bool InternalTryGetTypeHandle(out RuntimeTypeHandle runtimeTypeHandle)
```

Comment:
This is one of the pieces that makes things complicated - the higher level reflection code has assumptions that if we can get to a TypeHandle, the type has an EEType. This method is a compatibility shim and the uses of it will need further review.